frontend/taipy-gui/src/components/Taipy/ProgressBar.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import * as React from "react";
+import { useEffect, useState } from "react";
+import CircularProgress from "@mui/material/CircularProgress";
+import LinearProgress from "@mui/material/LinearProgress";
+import Box from "@mui/material/Box";
+
+interface ProgressBarProps {
+    linear?: boolean;
+    //showProgress?: boolean;
+}
+
+const Progress = (props: ProgressBarProps) => {
+    const { linear } = props;
+    const [value, setValue] = useState(false); //By default the circular element will be rendered. If the user declares the `linear` param as true, then the linear element will be rendered otherwise circular.
+
+    useEffect(() => {
```

Comment:
You could use `useMemo` instead of `useState` + `useEffect`

Reply:
Sure, I'll use `useMemo` and remove the `useState` and `useEffect`👍🏼

Reply:
Hey @FredLL-Avaiga, for using the `useMemo` hook. should I create a function inside of the `Progress` one named (let's say) - `ProgressElement`? What we can do is that inside this function, the element will be rendered and in the dependency array, the params can be - `[linear, showProgress, progressBarCount]`? Is this a good approach or should I change it?<br>
![image](https://github.com/Avaiga/taipy/assets/129659514/37883962-1bea-4239-a92e-19508490bf1c)

Reply:
You don't even need to use useMemo.

return <Box className={className}>
{props.linear ? <LinearProgress /> : <CircularProgress />}
</Box>

Reply:
Oh! ok👍🏼Actually in your previous comment, you suggested to use the `useMemo` hook, that's why I asked about it. But no issues, I'll do it without it as well👍🏼